eng/pipelines/templates/jobs/ci.yml

```diff
@@ -73,24 +73,22 @@ jobs:
         parameters:
           ServiceDirectory: ${{parameters.ServiceDirectory}}
 
-      - ${{ each artifact in parameters.Artifacts }}:
-        - script: |
-            python3 --version
-            python3 eng/versioning/set_versions.py --build-type ${{parameters.SDKType}} --build-qualifier alpha.$(Build.BuildNumber) --artifact-id ${{artifact.name}} --group-id ${{artifact.groupId}}
-          condition: and(succeeded(), eq(variables['SetDevVersion'],'true'))
-          displayName: Append dev package version suffix for ${{artifact.name}}
-
-      - script: |
+      - pwsh: |
+          $artifacts = '${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json
           python3 --version
+
+          # Append dev package version suffix for each artifact
+          foreach ($artifact in $artifacts) {
+            python3 eng/versioning/set_versions.py --build-type ${{parameters.SDKType}} --build-qualifier alpha.$(Build.BuildNumber) --artifact-id $artifact.name --group-id $artifact.groupId
+          }
+
+          # Set zero-dev-version for packages
           python3 eng/versioning/set_versions.py --set-dev-zero-version --build-type ${{parameters.SDKType}} --build-qualifier alpha.$(Build.BuildNumber)
-        condition: and(succeeded(), eq(variables['SetDevVersion'],'true'))
```

Comment:
This appears to have lost the guard on only being applied when `SetDevVersion == true`

Reply:
The diff looks bad but you can see it is still there on line 90 below. I merged 3 +number of artifact steps into a single step.